DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/DiagnosticTelemetryCOmmandProcessorTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Dynamic;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Configurations;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.Cooler.CommandProcessors;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.Cooler.Devices;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.CommandProcessors;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Logging;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Telemetry.Factory;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Transport;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Transport.Factory;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.UnitTests.Simulator.WebJob
+{
+    public class DiagnosticTelemetryCommandProcessorTests
+    {
+        public DiagnosticTelemetryCommandProcessorTests()
+        {
+            _loggerMock = new Mock<ILogger>();
+            _transportFactory = new Mock<ITransportFactory>();
+            _telemetryFactoryMock = new Mock<ITelemetryFactory>();
+            _configurationProviderMock = new Mock<IConfigurationProvider>();
+            _coolerDevice = new Mock<CoolerDevice>(_loggerMock.Object, _transportFactory.Object,
+                _telemetryFactoryMock.Object,
+                _configurationProviderMock.Object);
+            _diagnosticTelemetryCommandProcessor = new DiagnosticTelemetryCommandProcessor(_coolerDevice.Object);
+        }
+
+        private readonly Mock<CoolerDevice> _coolerDevice;
+        private readonly DiagnosticTelemetryCommandProcessor _diagnosticTelemetryCommandProcessor;
+        private readonly Mock<IConfigurationProvider> _configurationProviderMock;
+        private readonly Mock<ILogger> _loggerMock;
```

Comment:
if you're not using _loggerMock outside of constructor you can actually move the variable in there